File: JurosComposto/app/src/main/java/com/example/juroscomposto/MainActivity.java
package com.example.juroscomposto;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {
    private EditText txt_capital;
    private EditText txt_juros;
    private EditText txt_periodo;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        txt_capital = findViewById(R.id.txt_capital);
        txt_juros = findViewById(R.id.txt_juros);
        txt_periodo = findViewById(R.id.txt_periodo);
    }

    public void calcular(View view) {
        String txtCapital = txt_capital.getText().toString();
        String txtJuros = txt_juros.getText().toString();
        String txtPeriodo = txt_periodo.getText().toString();

        Intent intent = new Intent(this, CalculoActivity.class);

        intent.putExtra("txtCapital", txtCapital);
        intent.putExtra("txtJuros", txtJuros);
        intent.putExtra("txtPeriodo", txtPeriodo);

        startActivity(intent);
    }
}

File: MyTrip/app/src/main/java/com/example/mytrip/MainActivity.kt
package com.example.mytrip

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import java.lang.NumberFormatException

class MainActivity : AppCompatActivity(), View.OnClickListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // findViewById<>(R.id.editDistance) // R = resource => que é gerenciando pelo Android automaticamente
        // ir na build.grable e adicionar em plugins " id 'kotlin-android-extensions' "

        /*
        buttonCalculate.setOnClickListener {
            View.OnClickListener {
                println("teste")
            }
        }
        */

        buttonCalculate.setOnClickListener(this)
    }

    override fun onClick(view: View) {
        val id = view.id

        if (id == R.id.buttonCalculate) {
            handleCalculate()
        }
    }

    private fun handleCalculate() {
        if (isValid()) {

            try {
                val distance = editDistance.text.toString().toFloat()
                val price = editPrice.text.toString().toFloat()
                val autonomy = editAutonomy.text.toString().toFloat()

                val result = ((distance * price) / autonomy)
                textResult.setText("Total R$: $result")
            } catch (nfe: NumberFormatException) {
                Toast.makeText(this, getString(R.string.valores_validos), Toast.LENGTH_LONG).show()
            }

        } else {
            Toast.makeText(this, getString(R.string.valores_validos), Toast.LENGTH_LONG).show()
        }
    }

    private fun isValid(): Boolean {
        return editDistance.text.toString() != ""
                && editPrice.text.toString() != ""
                && editAutonomy.text.toString() != ""
                && editAutonomy.text.toString() != "0"

    }

}
